contracts/cryptoblades.sol

```diff
@@ -343,25 +351,29 @@ contract CryptoBlades is Initializable, AccessControlUpgradeable {
         uint16 xp = getXpGainForFight(playerFightPower, targetPower) * fightMultiplier;
         uint256 tokens = getTokenGainForFight(targetPower, true) * fightMultiplier;
 
-        if(playerRoll < monsterRoll) {
+        uint256 offset = (tokens.mul(combatTokenChargePercent).mul(tokensPrices.skillTokenPrice())).div(tokensPrices.tokenPrice());
```

Comment:
Still 2 external calls here. You can make it only one by creating a function in tokensPrices that will return skillTokenPrice()/tokenPrice() ratio.